src/QtLocationPlugin/QGeoTileFetcherQGC.cpp

```diff
@@ -88,3 +97,31 @@ void QGeoTileFetcherQGC::handleReply(QGeoTiledMapReply *reply, const QGeoTileSpe
         emit tileError(spec, reply->errorString());
     }
 }
+
+QNetworkRequest QGeoTileFetcherQGC::getNetworkRequest(int mapId, int x, int y, int zoom)
+{
+    const SharedMapProvider mapProvider = UrlFactory::getMapProviderFromQtMapId(mapId);
+
+    QNetworkRequest request;
+    request.setUrl(mapProvider->getTileURL(x, y, zoom));
+    request.setRawHeader(QByteArrayLiteral("Accept"), QByteArrayLiteral("*/*"));
+    request.setHeader(QNetworkRequest::UserAgentHeader, s_userAgent);
+    const QByteArray referrer = mapProvider->getReferrer().toUtf8();
+    if (!referrer.isEmpty()) {
+        request.setRawHeader(QByteArrayLiteral("Referrer"), referrer);
+    }
+    const QByteArray token = mapProvider->getToken();
+    if (!token.isEmpty()) {
+        request.setRawHeader(QByteArrayLiteral("User-Token"), token);
+    }
+    // request.setOriginatingObject(this);
+    request.setAttribute(QNetworkRequest::CacheLoadControlAttribute, QNetworkRequest::PreferCache);
+    request.setAttribute(QNetworkRequest::BackgroundRequestAttribute, true);
+    request.setAttribute(QNetworkRequest::CacheSaveControlAttribute, true);
+    request.setAttribute(QNetworkRequest::DoNotBufferUploadDataAttribute, false);
+    // request.setAttribute(QNetworkRequest::AutoDeleteReplyOnFinishAttribute, true);
```

Comment:
Did you plan to uncomment these or why are they commented?

Reply:
That's an option I was considering but needed to test more after remaining changes and didn't want to forget. It will either be deleted or uncommented by the end, but I can also just remove it now.